src/plugin/memory-webview-main.ts

```diff
@@ -93,8 +95,45 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
             vscode.commands.registerCommand(MemoryWebview.ShowCommandType, () => this.show()),
             vscode.commands.registerCommand(MemoryWebview.VariableCommandType, async args => {
                 if (isVariablesContext(args)) {
-                    const memoryReference = args.variable.memoryReference ?? await this.memoryProvider.getAddressOfVariable(args.variable.name);
-                    this.show({ memoryReference });
+                    let maybeError;
+                    /*
+                    Use the first available address from this list.
+                    1. The variable's memoryReference if it exists.
+                    2. Try to get the address for the variable name.
+                        2.1. Use the final component of the evaluateName, which handles arrays correctly (e.g. "my_char[2]")
+                        2.2. Fall back to using the simple name if there is no evaluateName
+                    3. If the container value appears to be a memory address, use that.
+                    4. Lastly, if the container has its own address, use that.
+                    */
```

Comment:
See above re: retaining current behavior of the existing 'view in Memory Inspector' command; could we move the other strategies into a dedicated command & handler for those use cases?